core/tests/protractor_desktop/voiceoverUploadFeatures.js

```diff
@@ -85,6 +89,52 @@ describe('Voiceover upload features', function() {
       '../data/cafe-over-five-minutes.mp3');
     explorationEditorTranslationTab.expectSaveUploadedAudioButtonToBeDisabled();
     explorationEditorTranslationTab.closeUploadAudioModal();
+    explorationEditorTranslationTab.deleteAudioRecord();
+    explorationEditorTranslationTab.confirmDeleteAudioRecord();
+  });
+
+  it('should upload recorded file', function() {
+    explorationEditorTranslationTab.addAudioRecord();
+    explorationEditorTranslationTab.stopAudioRecord();
+    explorationEditorTranslationTab.confirmAudioRecord();
+    explorationEditorTranslationTab.playAudioRecord();
+    browser.refresh();
+    explorationEditorTranslationTab.playAudioRecord();
+  });
+
+  it('should play recorded file after logging out', function() {
+    users.logout();
+    users.login(TEST_EMAIL);
+    creatorDashboardPage.get();
+    creatorDashboardPage.editExploration('Untitled');
+    explorationEditorMainTab.exitTutorial();
+
+    explorationEditorPage.navigateToTranslationTab();
+    explorationEditorTranslationTab.playAudioRecord();
+    explorationEditorTranslationTab.deleteAudioRecord();
+    explorationEditorTranslationTab.confirmDeleteAudioRecord();
+  });
+
+  it('should upload audio file from path', function() {
+    explorationEditorTranslationTab.uploadAudioRecord(
+      '../../../data/explorations/audio_test/assets/audio/test_audio_1_en.mp3');
+    explorationEditorTranslationTab.saveAudioRecord();
+    explorationEditorTranslationTab.playAudioRecord();
```

Comment:
Is it possible to check that the played audio file is the same as the one we uploaded? I'm not sure we need to do this; I'm just curious if you happen to know how hard it would be.

Reply:
After some searching, this [doesn't look easy](https://stackoverflow.com/a/17354617), so I think we can skip it.